frontend/src/pages/PipelineDetails.tsx

```diff
@@ -476,15 +576,51 @@ class PipelineDetails extends Page<{}, PipelineDetailsState> {
           graph: undefined,
           reducedGraph: undefined,
           graphV2,
-          selectedVersion,
+          v1SelectedVersion: selectedVersion,
           templateString: selectedVersionPipelineTemplate,
         });
       } else {
         this.setStateSafe({
           graph,
           reducedGraph,
           graphV2: undefined,
-          selectedVersion,
+          v1SelectedVersion: selectedVersion,
+          templateString: selectedVersionPipelineTemplate,
+        });
+      }
+    }
+  }
+
+  public async handleVersionSelectedV2(versionId: string): Promise<void> {
+    if (this.state.v2Pipeline) {
+      const selectedVersion = (this.state.v2Versions || []).find(
+        v => v.pipeline_version_id === versionId,
+      );
+      const selectedVersionPipelineTemplate = await this._getTemplateString(
+        this.state.v2Pipeline.pipeline_id!,
+        versionId,
+      );
+      this.props.history.replace({
+        pathname: `/pipelines/details/${this.state.v2Pipeline.pipeline_id}/version/${versionId}`,
+      });
+
+      const [graph, reducedGraph, graphV2] = await this._createGraph(
+        selectedVersionPipelineTemplate,
+      );
+      if (isFeatureEnabled(FeatureKey.V2_ALPHA) && graphV2.length > 0) {
```

Comment:
Question orthogonal to the PR: as we're going into GA, do we still need to support this feature key? 
/cc @zijianjoy